docs/src/pages/components/autocomplete/Virtualize.tsx

```diff
@@ -1,58 +1,64 @@
 import React from 'react';
 import TextField from '@material-ui/core/TextField';
-import Autocomplete from '@material-ui/lab/Autocomplete';
+import Autocomplete, { RenderGroupParams } from '@material-ui/lab/Autocomplete';
 import useMediaQuery from '@material-ui/core/useMediaQuery';
+import ListSubheader from '@material-ui/core/ListSubheader';
 import { useTheme, makeStyles } from '@material-ui/core/styles';
-import { FixedSizeList, ListChildComponentProps } from 'react-window';
+import { VariableSizeList, ListChildComponentProps } from 'react-window';
+import { Typography } from '@material-ui/core';
 
 function renderRow(props: ListChildComponentProps) {
   const { data, index, style } = props;
-
-  return React.cloneElement(data[index], {
-    style: {
-      overflow: 'hidden',
-      textOverflow: 'ellipsis',
-      whiteSpace: 'nowrap',
-      display: 'block',
-      ...style,
-    },
-  });
+  return React.cloneElement(data[index], { style });
 }
 
 // Adapter for react-window
-const ListboxComponent = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLElement>>(
-  function ListboxComponent(props, ref) {
-    const { children, ...other } = props;
-    const theme = useTheme();
-    const smUp = useMediaQuery(theme.breakpoints.up('sm'));
-    const itemCount = Array.isArray(children) ? children.length : 0;
-    const itemSize = smUp ? 36 : 48;
+const ListboxComponent = React.forwardRef<HTMLDivElement>(function ListboxComponent(props, ref) {
+  const { children, ...other } = props;
+  const itemData = React.Children.toArray(children);
+  const theme = useTheme();
+  const smUp = useMediaQuery(theme.breakpoints.up('sm'), { noSsr: true });
+  const itemCount = itemData.length;
+  const itemSize = smUp ? 36 : 48;
 
-    const outerElementType = React.useMemo(() => {
-      return React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
-        (props2, ref2) => <div ref={ref2} {...props2} {...other} />,
-      );
-    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+  const getChildSize = (child: React.ReactNode) => {
+    if (React.isValidElement(child) && child.type === ListSubheader) {
+      return 48;
+    }
 
-    return (
-      <div ref={ref}>
-        <FixedSizeList
-          style={{ padding: 0, height: Math.min(8, itemCount) * itemSize, maxHeight: 'auto' }}
-          itemData={children}
-          height={250}
-          width="100%"
-          outerElementType={outerElementType}
-          innerElementType="ul"
-          itemSize={itemSize}
-          overscanCount={5}
-          itemCount={itemCount}
-        >
-          {renderRow}
-        </FixedSizeList>
-      </div>
-    );
-  },
-);
+    return itemSize;
+  };
+
+  const getHeight = () => {
+    if (itemCount > 8) {
+      return 8 * itemSize;
+    }
+    return itemData.map(getChildSize).reduce((a, b) => a + b, 0);
```

Comment:
What if there are 1,000 items in a group?

Reply:
`itemCount` would be greater than 8?

Reply:
So https://codesandbox.io/s/virtualized-grouped-attempt-wm0hi doesn't handle it correctly?

Reply:
Only under the `sm` breakpoint. Because there the subheaders have the same size as the items. Following gif illustrates the problem, note how the window tries to size to two items, but the larger subheader makes it overflow:
![mui](https://user-images.githubusercontent.com/2109932/70480408-eded4300-1adf-11ea-8de3-d47f7250aac0.gif)
My version looks like:
![mui2](https://user-images.githubusercontent.com/2109932/70480569-5b996f00-1ae0-11ea-8bc5-ab49d8105dd7.gif)

Reply:
@oliviertassinari Maybe the subheaders need to be responsive and shrink to 36px on desktop as well?

Reply:
Ok, this looks good :+1:. I'm not sure about the list, this height change is specific to the menu. I think that we will need to revamp this point in v5.